### PR TITLE
Draft v0.8.0 release notes

### PR DESCRIPTION
**Pull request by leewujung**
Draft v0.7.2 release notes in doc What's new section.

@lsetiawan @emiliom : I cherry picked items from the PR comments that are more relevant to the users in the release notes. Feel free to propose changes!

### CODE REVIEW

**Review by lsetiawan — APPROVED** [review]
LGTM

**Comment by leewujung** [comment]
We've decided to make the upcoming release 0.8.0 due to breaking changes in the data format, in particular the dropping of redundant `beam` and `ping_time` dimensions from some data variables. I've updated the issue title.

File: docs/source/whats-new.md [hunk 1]
```diff
@@ -4,6 +4,44 @@ What's new
 See [GitHub releases page](https://github.com/OSOceanAcoustics/echopype/releases) for the complete history.
 
 
+# v0.7.2 (2023 July 14)
```
Comment (lsetiawan) [inline docs/source/whats-new.md:7]:
```suggestion
# v0.8.0 (2023 July 14)
```

Should probably also change the date from `July 14`

Comment (leewujung) [reply docs/source/whats-new.md:7]: Yes! it depends on #1056 and #1057

**Comment by leewujung** [comment]
This is updated with all commits up to Aug 21. Will add any new merges to `dev` right before the release.

File: docs/source/whats-new.md [hunk 2]
```diff
@@ -4,6 +4,63 @@ What's new
...
+
+## Enhancement
+- Overhaul the `combine_echodata` function (#1042, #1116)
+  - No longer require spinning up dask client under the hood during combine
+  - Use `xr.concat` directly compatible with delayed computation
+  - No longer require destination path as input argument
+- Overhaul `EchoData.update_platform` (#1060)
+  - Allow adding external variables with arbitrary number of time dimensions, with new time dimensions following an increasing order (e.g., `time3`, `time4`, etc)
+  - Only allow updating standardized, sonar model-dependent Platform group variables
+- Remove outdated `ui_param` (#1071)
```
Comment (emiliom) [inline docs/source/whats-new.md:21]: I'd move this to the "Others" section

Comment (leewujung) [reply docs/source/whats-new.md]: Done.

**Pull request merged**